lib/src/transport_platform.dart

```diff
@@ -82,9 +82,11 @@ class MockAwareTransportPlatform {
       @Deprecated(v3Deprecation) List<String> sockJSProtocolsWhitelist,
       @Deprecated(v3Deprecation) Duration sockJSTimeout,
       @Deprecated(v3Deprecation) bool useSockJS}) {
-    if (MockTransportsInternal.isInstalled) {
+    if (MockTransportsInternal.isInstalled &&
+        MockWebSocketInternal.hasHandlerForWebSocket(uri)) {
       return MockWSocket.connect(uri, headers: headers, protocols: protocols);
-    } else if (realTransportPlatform != null) {
+    } else if (MockTransportsInternal.fallThrough &&
+        realTransportPlatform != null) {
```

Comment:
Added some missing checks here - this ensures two things:

1. A mock-aware websocket is only returned if mocks are installed _and_ there is a handler for it.
2. A real websocket is only returned if fallThrough is turned on